velox/functions/CoverageUtil.h

```diff
@@ -0,0 +1,138 @@
+
+class TablePrinter {
+ public:
+  TablePrinter(
+      size_t numScalarColumns,
+      size_t columnSize,
+      std::string indent,
+      std::ostream& out)
+      : numScalarColumns_{numScalarColumns},
+        columnSize_{columnSize},
+        indent_{std::move(indent)},
+        out_{out} {}
+
+  void header() {
+    std::string line(columnSize_, '=');
+    out_ << indent_ << line;
+    for (int i = 1; i < numScalarColumns_; i++) {
+      out_ << "  " << line;
+    }
+    out_ << "  ==  " << line << std::endl;
+
+    auto scalarFunctionsColumnWidth =
+        columnSize_ * numScalarColumns_ + 2 * (numScalarColumns_ - 1);
+
+    out_ << indent_ << std::left << std::setw(scalarFunctionsColumnWidth)
+         << "Scalar Functions"
+         << "      "
+         << "Aggregate Functions" << std::endl;
+    out_ << indent_ << std::string(scalarFunctionsColumnWidth, '=')
+         << "  ==  " << line << std::endl;
+  }
+
+  void startRow() {
+    out_ << indent_;
+    currentColumn_ = 0;
+  }
+
+  void addColumn(const std::string& text) {
+    addColumn(text, columnSize_);
+  }
+
+  void addEmptyColumn() {
+    if (currentColumn_ == numScalarColumns_) {
+      addColumn("", 2);
+    } else {
+      addColumn("", columnSize_);
+    }
+  }
+
+  void endRow() {
+    out_ << std::endl;
+  }
+
+  void footer() {
+    std::string line(columnSize_, '=');
+    out_ << indent_ << line;
+    for (int i = 1; i < numScalarColumns_; i++) {
+      out_ << "  " << line;
+    }
+    out_ << "  ==  " << line << std::endl;
+  }
+
+ private:
+  void addColumn(const std::string& text, size_t columnSize) {
+    if (currentColumn_) {
+      out_ << "  ";
+    }
+    out_ << std::setw(columnSize) << text;
+    ++currentColumn_;
+  }
+
+  const size_t numScalarColumns_;
+  const size_t columnSize_;
+  const std::string indent_;
+  std::ostream& out_;
+
+  size_t currentColumn_{0};
+};
+
+class TableCellTracker {
+ public:
+  // Takes zero-based row and column numbers.
+  void add(int row, int column) {
+    cells_.emplace_back(row, column);
+  }
+
+  const std::vector<std::pair<int, int>>& cells() const {
+    return cells_;
+  }
+
+ private:
+  std::vector<std::pair<int, int>> cells_;
+};
+
+void printCoverageMap(
+    const std::vector<std::string>& scalarNames,
+    const std::vector<std::string>& aggNames,
+    const std::unordered_set<std::string>& veloxNames,
+    const std::unordered_set<std::string>& veloxAggNames,
+    const std::string& domain = "");
+
+/// Takes a super-set of simple, vector and aggregate function names and prints
+/// coverage map showing which of these functions are available in Velox.
+void printCoverageMap(
+    const std::vector<std::string>& scalarNames,
+    const std::vector<std::string>& aggNames,
+    const std::string& domain = "");
+
+void printCoverageMapForAll(const std::string& domain = "");
+
+void printVeloxFunctions(
+    const std::unordered_set<std::string> kLinkBlockList,
```

Comment:
const std::unordered_set<...>& linkBlockList

Reply:
Just fixed it.